scripts/local-npm.sh

```diff
@@ -0,0 +1,152 @@
+  export PATH="$thisdir:$PATH"
+
+  # Kill `verdaccio` when the script exits
+  trap 'kill $(cat "$HOME/verdaccio.pid")' EXIT
+
+  (
+    cd "$HOME"
+    echo "Starting verdaccio in background..."
+    : > verdaccio.log
+    nohup npx verdaccio@^5.4.0 &>verdaccio.log &
+    echo $! > verdaccio.pid
+
+    # Wait for `verdaccio` to boot
+    grep -q 'http address' <(tail -f verdaccio.log)
+
+    # Set registry to local registry
+    npm set registry http://localhost:4873
+    yarn config set registry http://localhost:4873
+
+    # Login so we can publish packages
+    npx npm-cli-login@^1.0.0 -u user -p password -e user@example.com \
+      -r http://localhost:4873 --quotes
+
+    npm whoami
+  ) 1>&2
+}
+
+publish() {
+  export DISTTAG=agblah
+  case $1 in
+  manual*)
+    flags=--no-git-tag-version
+    ;;
+  *)
+    flags=--canary
+    ;;
+  esac
+
+  # Publish the packages to our local service.
+  # without concurrency until https://github.com/Agoric/agoric-sdk/issues/8091
+  yarn lerna publish --concurrency 1 --conventional-prerelease $flags --exact \
+    --dist-tag="$DISTTAG" --preid=dev"-$(git rev-parse --short=7 HEAD)" \
+    --no-push --no-verify-access --yes
+
+  # Use the locally-installed dist-tag.
+  persistVar LOCAL_NPM_HOME "$HOME"
+  persistVar LOCAL_NPM_DISTTAG "$DISTTAG"
+}
+
+integrationTest() {
+  # Install the Agoric CLI on this machine's $PATH.
+  case $1 in
+  link-cli)
+    yarn link-cli "$HOME/bin/agoric"
+    persistVar AGORIC_CMD "[\"$HOME/bin/agoric\"]"
+    ;;
+  */npm)
+    npm install -g "agoric@$DISTTAG" </dev/null
+    persistVar AGORIC_CMD '["agoric"]'
+    ;;
+  */npx)
+    # Install on demand.
+    persistVar AGORIC_CMD "[\"npx\",\"agoric@$DISTTAG\"]"
+    ;;
+  *)
+    yarn global add "agoric@$DISTTAG" </dev/null
+    persistVar AGORIC_CMD '["agoric"]'
+    ;;
+  esac
+
+  test -z "${DISTTAG-}" || {
+    persistVar AGORIC_INSTALL_OPTIONS "[\"$DISTTAG\"]"
+  }
+  persistVar AGORIC_START_OPTIONS '["--rebuild"]'
+  persistVar AGORIC_INIT_OPTIONS "[\"--dapp-branch=$2\"]"
+
+  (
+    cd "$thisdir/../packages/agoric-cli"
+
+    # Try to avoid hitting a pessimal Actions output rate-limitation.
+    SOLO_MAX_DEBUG_LENGTH=1024 \
+      yarn integration-test
+  )
+}
+
+case ${1-} in
+ci)
+  verdaccio </dev/null
+  publish "${2-manual}"
+  integrationTest "${2-manual}" "${3-main}"
+  ;;
+
+bg)
+  verdaccio </dev/null
+  echo "Publish packages using '$0 publish'"
+  trap - EXIT
+  echo "HOME=$HOME"
+  echo "pid=$(cat "$HOME/verdaccio.pid")"
+  ;;
+
+bg-publish)
+  verdaccio
+  trap - EXIT
+  publish "${2-local-npm}"
+
+  # Git dirty check.
+  if [[ -n "$(git status --porcelain)" ]]; then
+    echo "Git status is dirty, aborting."
+    git status
+    exit 1
+  fi
+  ;;
+
+publish)
+  publish "${2-manual}"
+  echo "Run getting-started integration test with '$0 test'"
+  ;;
+
+test)
+  test -z "${LOCAL_NPM_HOME-}" || export HOME="$LOCAL_NPM_HOME"
+  test -z "${LOCAL_NPM_DISTTAG-}" || export DISTTAG="$LOCAL_NPM_DISTTAG"
```

Comment:
not clear to me why not: HOME = HOME || LOCAL_NPM_HOME.

Reply:
I actually want `LOCAL_NPM_HOME && (HOME=LOCAL_NPM_HOME)` but I can't find a more concise way to write it.